pkg/operator/rules.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/model/rulefmt"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	thanostypes "github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+func GenerateRulesConfigurationForPrometheus(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
+	// Unset partialResponseStrategy field.
+	for i := range promRule.Groups {
+		promRule.Groups[i].PartialResponseStrategy = ""
+	}
+
+	return generateRulesConfiguration(promRule, logger)
+}
+
+func GenerateRulesConfigurationForThanos(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
```

Comment:
in the same vein, `GenerateRulesConfigurationForThanos` could go away and pkg/thanos can call `operator.GenerateRulesConfiguration(...)` directly.